WolvenKit/Views/Editors/AutomaticNodifyEditor.cs

```diff
@@ -0,0 +1,232 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using Nodify;
+using Microsoft.Msagl.Core.Geometry.Curves;
+using Microsoft.Msagl.Core.Layout;
+using Microsoft.Msagl.Core.Routing;
+using Microsoft.Msagl.Layout.Layered;
+using WolvenKit.Functionality.Interfaces;
+using System.Windows.Input;
+using Microsoft.Msagl.Layout.MDS;
+
+namespace WolvenKit.Views.Editors
+{
+    public class AutomaticNodifyEditor : NodifyEditor
+    {
+        public AutomaticNodifyEditor() : base()
+        {
+
+        }
+
+
+        protected override void OnMouseMove(MouseEventArgs e)
+        {
+            CurrentMousePosition = e.GetPosition(this);
+
+            if (CurrentMousePosition != PreviousMousePosition)
+            {
+                // All coordinates that the editor works with should be inside the ItemsHost. Translating coordinates is not needed if,
+                // for example, you want to move the viewport to a certain location inside the editor's space. (like focusing a node)
+                MouseLocation = e.GetPosition(ItemsHost);
+                if (IsMouseCaptured)
+                {
+                    // Panning
+                    if (e.MiddleButton == MouseButtonState.Pressed)
+                    {
+                        // Need to scale this by the zoom since the ItemsHost changes its transform based on the mouse delta
+                        ViewportLocation -= (CurrentMousePosition - PreviousMousePosition) / ViewportZoom;
+                        IsPanning = true;
+                        e.Handled = true;
+                    }
+                    else if (IsSelecting)
+                    {
+                        Selection.Update(MouseLocation);
+                    }
+                    else
+                    {
+                        // Should not reach this
+                        ReleaseMouseCapture();
+                    }
+                }
+
+                PreviousMousePosition = CurrentMousePosition;
+            }
+        }
+
+        protected override void OnMouseRightButtonDown(MouseButtonEventArgs e)
+        {
+
+        }
+
+        protected override void OnMouseRightButtonUp(MouseButtonEventArgs e)
+        {
+
+        }
+
+        protected override void OnMouseDown(MouseButtonEventArgs e)
+        {
+            if (e.ChangedButton == MouseButton.Middle)
+            {
+                if (Mouse.Captured == null)
+                {
+                    Focus();
+                    CaptureMouse();
+
+                    InitialMousePosition = e.GetPosition(this);
+                }
+            }
+            else
+            {
+                base.OnMouseDown(e);
+            }
+        }
+
+        protected override void OnMouseUp(MouseButtonEventArgs e)
```

Comment:
Hi! I just released **v4.0.0** to address the configurable mouse/key bindings issue https://github.com/miroiu/nodify/issues/46. Updating to this version should remove the need to override mouse event handlers. You can have a look at [EditorGestures](https://github.com/miroiu/nodify/blob/master/Nodify/EditorGestures.cs) to see what can be configured and how. And you can also create custom interactions if needed by overriding the `NodifyEditor.GetInitialState`.

Here's the [changelog](https://github.com/miroiu/nodify/blob/master/CHANGELOG.md) 🚀